app/scripts/modules/google/address/address.reader.ts

```diff
@@ -0,0 +1,60 @@
+import {module, IPromise} from 'angular';
+import {SEARCH_SERVICE, SearchService, ISearchResults} from 'core/search/search.service';
+import {InfrastructureCacheService, INFRASTRUCTURE_CACHE_SERVICE} from 'core/cache/infrastructureCaches.service';
+
+interface IAddressSearchResults {
+  account: string;
+  address: string; // JSON encoded string containing the address. It is missing the account.
+  name: string;
+  provider: string;
+  region: string;
+  type: string;
+}
+
+export interface IGceAddress {
+  account: string;
+  address: string;
+  creationTimestamp: string;
+  description: string;
+  id: number;
+  kind: string;
+  labelFingerprint: string;
+  name: string;
+  networkTier: string;
+  region: string;
+  selfLink: string;
+  status: string;
+}
+
+class GceAddressReader {
+  static get $inject() { return ['searchService', 'infrastructureCaches']; }
+
+  constructor (private searchService: SearchService, private infrastructureCaches: InfrastructureCacheService) {}
+
+  public listAddresses(region?: string): IPromise<IGceAddress[]> {
+    if (region) {
+      return this.listAddresses().then(addresses => addresses.filter(address => address.region === region));
```

Comment:
Can you add an explicit null and a comment in the recursive `listAddresses` to make it a bit clearer? Something like `listAddresses(null /* region */)` to make this a little less subtle.